src/backend/backend.ts

```diff
@@ -214,31 +205,20 @@ const updateRouterTree = (): Promise<void> => {
       updateLazyLoadedNgModules(routers);
       messageBuffer.enqueue(MessageFactory.routerTree(parsedRoutes));
     }
-
   });
 };
 
-const subject = new Subject<void>();
+let ngModuleRef: NgModuleRef<any>;
+let isStableSubscription: Subscription;
 
-const subscriptions = new Array<Subscription>();
+const collectRoots = () => getAllAngularRootElements().map(r => ng.probe(r)).filter(x => x !== null);
 
-const bind = (root) => {
-  const ngZone = root.injector.get(ng.coreTokens.NgZone);
-  if (ngZone) {
-    subscriptions.push(ngZone.onStable.subscribe(() => subject.next(void 0)));
-  }
-
-  // parse components and routes each time
-  subscriptions.push(
-    subject.debounceTime(0).subscribe(() => {
-      Promise.all([
-        updateComponentTree(getAllAngularRootElements().map(r => ng.probe(r))),
-        updateRouterTree()
-      ]).then(() => onUpdateNotifier.next());
-    }));
-
-  // initial load
-  subject.next(void 0);
+const listenForSomeTimeAndMaybeResubscribe = (timeMs: number) => {
```

Comment:
Any chance something like this approach might work in `augury-labs` lazy loaded module support?

Reply:
@andrewthauer Possibly, but it's all still a little hazy for now.